src/providers/v2v3/NftRewardsProvider.tsx

```diff
@@ -6,31 +6,37 @@ import { useNftRewardTiersOf } from 'hooks/JB721Delegate/contractReader/NftRewar
 import useNftRewards from 'hooks/NftRewards'
 import { useContext } from 'react'
 import { EMPTY_NFT_COLLECTION_METADATA } from 'redux/slices/editingV2Project'
-import { CIDsOfNftRewardTiersResponse } from 'utils/nftRewards'
+import { CIDsOfNftRewardTiersResponse, hasNftRewards } from 'utils/nftRewards'
 
 export const NftRewardsProvider: React.FC = ({ children }) => {
   const { fundingCycleMetadata } = useContext(V2V3ProjectContext)
   const { projectMetadata, projectId } = useContext(ProjectMetadataContext)
 
   const dataSourceAddress = fundingCycleMetadata?.dataSource
+  // don't fetch stuff if there's no datasource in the first place.
+  const shouldFetch = hasNftRewards(fundingCycleMetadata)
 
   /**
    * Load NFT Rewards data
    */
   const { data: nftRewardTiersResponse, loading: nftRewardsCIDsLoading } =
     useNftRewardTiersOf({
-      dataSourceAddress: dataSourceAddress,
+      dataSourceAddress,
+      shouldFetch,
     })
+  // catchall to ensure nfts are never loaded if hasNftRewards is false (there's no datasource).
+  const tierData = shouldFetch ? nftRewardTiersResponse ?? [] : []
```

Comment:
this is the actual fix. even if dataSource is 0 or undefined, there's some sketchy race condition or cache issue where `nftRewardTiersResponse` is still defined sometimes.

This line will set it to [] if there's no datasource (failsafe, hopefully).

Reply:
Nice ok, good failsafe